packages/jellyfish-api-core/__tests__/category/loan/tokenSplits.test.ts

```diff
@@ -0,0 +1,966 @@
+    expect(vault3.state).toStrictEqual('active')
+
+    const vault4 = await testing.rpc.vault.getVault(vaultId4)
+    expect(vault4.state).toStrictEqual('active')
+  }
+
+  // Going to liquidate the vaults by price increase of the loan tokens
+  await testing.rpc.oracle.setOracleData(oracleID, timestamp, {
+    prices: [{
+      tokenAmount: '2.2@AAPL',
+      currency: 'USD'
+    }]
+  })
+  await testing.generate(1)
+  await testing.container.waitForActivePrice('AAPL/USD', '2.2')
+  await testing.rpc.oracle.setOracleData(oracleID, timestamp, {
+    prices: [{
+      tokenAmount: '2.2@TSLA',
+      currency: 'USD'
+    }]
+  })
+  await testing.container.waitForActivePrice('TSLA/USD', '2.2')
+
+  const [auction1, auction2] = await testing.rpc.vault.listAuctions()
+  {
+    const vault1 = await testing.rpc.vault.getVault(auction1.vaultId)
+    expect(vault1.state).toStrictEqual('inLiquidation')
+
+    const vault2 = await testing.rpc.vault.getVault(auction2.vaultId)
+    expect(vault2.state).toStrictEqual('inLiquidation')
+
+    const vault3 = await testing.rpc.vault.getVault(vaultId3)
+    expect(vault3.state).toStrictEqual('active')
+
+    const vault4 = await testing.rpc.vault.getVault(vaultId4)
+    expect(vault4.state).toStrictEqual('active')
+  }
+
+  await testing.rpc.account.sendTokensToAddress({}, { [collateralAddress]: ['252@AAPL'] })
+  await testing.generate(1)
+
+  const txid = await testing.container.call('placeauctionbid', [vaultId1, 0, collateralAddress, '5252@AAPL'])
+  expect(typeof txid).toStrictEqual('string')
+  expect(txid.length).toStrictEqual(64)
+  await testing.generate(1)
+}
+
+async function setupFutureSwap (): Promise<void> {
+  // Futures setup
+  // set the dfip2203/active to false
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'false' } })
+  await testing.generate(1)
+
+  // set dfip2203 params
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/reward_pct': `${futRewardPercentage}`, 'v0/params/dfip2203/block_period': `${futInterval}` } })
+  await testing.generate(1)
+
+  // set the dfip2203/active to true
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'true' } })
+  await testing.generate(1)
+
+  // Retrieve and verify gov vars
+  const attributes = await testing.rpc.masternode.getGov(attributeKey)
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/active']).toStrictEqual('true')
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/reward_pct']).toStrictEqual(`${futRewardPercentage}`)
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/block_period']).toStrictEqual(`${futInterval}`)
+}
+
+async function checkTokenSplit (tokenID: string, tokenSymbol: string, tokenSuffix: string, minted: BigNumber, checkLoan: boolean, checkCollateral: boolean): Promise<string> {
+  // check old token
+  const oldToken = (await testing.rpc.token.getToken(tokenID))[tokenID]
```

Comment:
hmm.. is the `oldToken` still functional?? 
since it still can be got by id.. not sure it can still be updated.. :thinking:

Reply:
no. It's finalized.